### ISSUE TITLE
`match_image` seems to keep pixel id

### ISSUE THREAD

**Issue by edarin** [goclem/delineation#2]
I needed an additional step to remove from the output the unlivable pixels that kept the pixel id from reference.

match_image <- function(density, reference, livable) {
  density <- density[, .(density = sum(density)), by = pixel]
  density <- density[match(reference[livable], pixel), "density"]
  density <- density[is.na(density), density := 0]
  image   <- replace(reference, livable, density$density)
  **image  <- replace(image, !livable, 0)**
  return(image)
}

https://github.com/goclem/delineation/blob/b02e6b9a2cb04f54759021cd35b4cea29ec5d08d/delineation_building.R#L155

**Comment by edarin** [comment]
Happy to add those as PR but was not sure how/if you want people to contribute.

**Comment by goclem** [comment]
Of course, comments/suggestions/corrections are more than welcome! I'm not sure to understand what additional step is needed however.

**Comment by edarin** [comment]
I've put it as last line in the function: ` image <- replace(image, !livable, 0)`, ie to replace the pixel_id values in reference that are in the unlivable area.
